caches_bench_test.go

```diff
@@ -47,60 +49,75 @@ func (sc structConstructor) Get(n int) myStruct {
 	return myStruct{Id: n}
 }
 
-func (sc structConstructor) Parse(data []byte) (myStruct, error) {
-	var s myStruct
-	err := json.Unmarshal(data, &s)
-	return s, err
+func (sc structConstructor) Parse(data []byte) myStruct {
+	return myStruct{Id: int(binary.BigEndian.Uint64(data))}
 }
 
-func (sc structConstructor) ToBytes(v myStruct) ([]byte, error) {
-	return json.Marshal(v)
+func (sc structConstructor) ToBytes(v myStruct) []byte {
+	b := [8]byte{}
+	binary.BigEndian.PutUint64(b[:], uint64(v.Id))
+	return b[:]
 }
 
 func MapSet[T any](cs constructor[T], b *testing.B) {
+	m := make(map[string]T, maxEntryCount)
+
+	id := rand.Intn(maxEntryCount)
 	for i := 0; i < b.N; i++ {
-		m := make(map[string]T, maxEntryCount)
-		for n := 0; n < maxEntryCount; n++ {
-			m[key(n)] = cs.Get(n)
+		if id >= maxEntryCount {
```

Comment:
`id = (id + 1) % maxEntryCount` ?

Reply:
You know that modulo is an IDIV operation which is one of the slowest CPU instructions on x86 (and other architectures as well)!? It can take up to 100 CPU cycles or so, while regular instructions can be down to 0.2 cycles. I try to avoid % and / in tight loops if possible.

Reply:
I know but I bet that this op will dominate in the benchmark. + main idea of any benchmark is to be fair, if every benchmark func will do modulo this will not change benchmark result (raw numbers might be different but the ratio will remain the same 🤷 )

Reply:
Whatever, feel free to leave as it is, I just proposed a shorter version 😉